include/sparrow/optional.hpp

```diff
@@ -121,6 +121,14 @@ namespace sparrow
         static constexpr bool is_optional_v = is_optional<T>::value;
     }
 
+    /**
+     * The optional class is similar to std::optional, with two major differences:
+     * - it can act as a proxy, meaning its template parameter can be lvalue references
+     *   or lvalue const references
+     * - the semantic for empty optoinal: resetting an non empty optional does not destruct
+     *   the contained value. Allocating an empty optional construct the contained value. This
+     *   behavior is temporary and will be changed in the near future.
+     */
     template <class T, class B = bool>
     class optional
```

Comment:
```suggestion
    template <class T, class B = bool>
        requires std::constructible_from<B, bool>
    class optional
```

Reply:
It also requires to be convertible to bool Let's create a concept and use it in front of the template parameter, otherwise we need to implement the class in its definition.